fix(app): Show module setup screen on the Desktop app for the given module.

vegano1 opened the pull request:
# Overview

Show the "Setup Module" screen when setting up a module on the Desktop app.
Closes: [RQA-4208](https://opentrons.atlassian.net/browse/RQA-4208)

<img width="1025" alt="Screenshot 2025-06-02 at 2 42 12 PM" src="https://github.com/user-attachments/assets/5310f603-94c2-42b5-99b2-000c9fba8bf1" />

<img width="1024" alt="Screenshot 2025-06-02 at 2 50 27 PM" src="https://github.com/user-attachments/assets/d743de2d-2a60-4bba-bac0-858d4dc4726f" />

## Test Plan and Hands on Testing

- [x] Make sure that clicking on the "Setup Module" notification on the desktop app shows the "Setup Module" screen for the given module, even if there are multiple modules that need to be set up.
- [x] Make sure the "Success" screen at the end of the flow does NOT show the "Setup another module" button for the desktop app.
- [x] Make sure the "Setup module" toast on the ODD is not shown if a run is in progress
- [x] Make sure the changes dont affect protocol setup, etc

## Changelog

- Show the "Setup Module" screen on the desktop app for the given module.
- Don't show the "Setup another module" button in module setup for the desktop app.
- Don't show the "Setup module" toast on the ODD if a run is in progress.

## Review requests

- Does this make sense?

## Risk assessment

Low, unreleased

[RQA-4208]: https://opentrons.atlassian.net/browse/RQA-4208?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

vegano1 committed:
fix(app): various fixes and improvements to module setup flow
- add setup another module button + logic to success screen
- fix some screens that werent using the correct SmallButton vs PrimaryButton
- blink module when we load the module setup launcher screen
- add useSendIdentifyModule to new utils.ts file so we can reuse createLiveCommands
- blink module red if install detection fails during module setup launcher
- blink module blue when the device re-connects after a firmware update
- stop blinking module when module setup is complete and the modal exits

vegano1 committed:
update SimpleWizardBody to allow for null children, so we can conditionally render them

vegano1 committed:
dont do dumb useEffect things :)

vegano1 committed:
- rename utils.ts -> hooks.tsx
- block moduleSetup exit when module is updating
- stop blinking module if the flow is closed while in the SelectModule component
- show the New Module Setup screen when launching module setup from the ModuleCard component
- correctly start/stop blinking when retrying after InstallDetect failure

vegano1 committed:
- Bring back attachedModule to ModuleCard, will address in separate pr
- Make sure we are correctly changing the LED state when setting up multiple modules

vegano1 committed:
fix(app): Conditionally show ModuleSelect component

vegano1 committed:
save

vegano1 committed:
dont show select module screen by default

vegano1 committed:
add showSetupLauncher to onDeviceDisplay

vegano1 committed:
use correct button for UpdateFirmware component

vegano1 committed:
Merge branch 'EXEC-1549-setup-another-module' into RQA-4208-show-module-attached-screen

vegano1 committed:
increase module update timeout from 30s to 60s

vegano1 committed:
Merge branch 'EXEC-1549-setup-another-module' into RQA-4208-show-module-attached-screen

vegano1 committed:
revert exitNow

vegano1 committed:
format

ahiuchingau reviewed (approved):
Nice thank you!

TamarZanzouri reviewed (approved):
Nice!

sfoster1 reviewed (changes requested):
I don't think we should add a current run poll to the top level like that - we're really trying to cut down on those. What if we instead check to see if there's a current run only when we recieve a module-connected event?

vegano1 committed:
- make components into functions instead of const
- dont return optional Null from JSX component, let the caller deal with conditonal rendering
- dont define react components constants mid code, then use them in the return (render) function
- replace console.error usage with setErrorMessage as these errors are non-recoverable for this flow + lets us get rid of returning null

vegano1 committed:
cleanup

vegano1 committed:
Merge branch 'edge' into EXEC-1549-setup-another-module

vegano1 committed:
include isSingleModule in useEffect

vegano1 committed:
Merge branch 'EXEC-1549-setup-another-module' into RQA-4208-show-module-attached-screen

vegano1 committed:
check if null

sfoster1 reviewed (approved):
All right, sounds good to me!

vegano1 committed:
Merge branch 'edge' into RQA-4208-show-module-attached-screen
